<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2026-02-06" />

<title>figure 1</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">autism_epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/han16/autism_epigenetics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">figure 1</h1>
<h4 class="date">2026-02-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2026-02-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>autism_epigenetics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20251118code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20251118)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20251118code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20251118)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhan16autismepigeneticstree5e48e7e7886700315eec5203bc459f774e73f1batargetblank5e48e7ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/han16/autism_epigenetics/tree/5e48e7e7886700315eec5203bc459f774e73f1ba" target="_blank">5e48e7e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhan16autismepigeneticstree5e48e7e7886700315eec5203bc459f774e73f1batargetblank5e48e7ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/han16/autism_epigenetics/tree/5e48e7e7886700315eec5203bc459f774e73f1ba" target="_blank">5e48e7e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Unstaged changes:
    Modified:   analysis/figure1_analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/figure1_analysis.Rmd</code>) and
HTML (<code>docs/figure1_analysis.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/autism_epigenetics/blob/5e48e7e7886700315eec5203bc459f774e73f1ba/analysis/figure1_analysis.Rmd" target="_blank">5e48e7e</a>
</td>
<td>
han
</td>
<td>
2026-02-09
</td>
<td>
2/9/2026
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/autism_epigenetics/5e48e7e7886700315eec5203bc459f774e73f1ba/docs/figure1_analysis.html" target="_blank">5e48e7e</a>
</td>
<td>
han
</td>
<td>
2026-02-09
</td>
<td>
2/9/2026
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/autism_epigenetics/98e85d668473b2bc884aa603ecd396332aa110c6/docs/figure1_analysis.html" target="_blank">98e85d6</a>
</td>
<td>
han
</td>
<td>
2026-02-06
</td>
<td>
2/6/2026
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/autism_epigenetics/blob/85aa352d46f4a180cef7237d1179c4c86f7c4fc8/analysis/figure1_analysis.Rmd" target="_blank">85aa352</a>
</td>
<td>
han
</td>
<td>
2026-02-06
</td>
<td>
2/6/2026
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/autism_epigenetics/85aa352d46f4a180cef7237d1179c4c86f7c4fc8/docs/figure1_analysis.html" target="_blank">85aa352</a>
</td>
<td>
han
</td>
<td>
2026-02-06
</td>
<td>
2/6/2026
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>rm(list=ls())
library(rprojroot)
library(dplyr)
root &lt;- rprojroot::find_rstudio_root_file()</code></pre>
<pre class="r"><code>union_peak_reads_count=read.delim((file.path(root, &quot;../Data/Novogene/PeakWindows/Cere_union_peak_read_counts.tsv&quot;)), header=T, stringsAsFactors = FALSE)
dim(union_peak_reads_count)</code></pre>
<pre><code>[1] 237934      9</code></pre>
<pre class="r"><code>head(union_peak_reads_count)</code></pre>
<pre><code>   chr   start     end C1 C2 C_merged K1 K2 K_merged
1 chr1 3083877 3084023 21  7       28  6  2        8
2 chr1 3120283 3120658 62 78      140 41 67      108
3 chr1 3128676 3128999  4  4        8 23 30       53
4 chr1 3149871 3150061 31 27       58  7 12       19
5 chr1 3251199 3251666 61 50      111 18 53       71
6 chr1 3300549 3300826 42 22       64 13  8       21</code></pre>
<pre class="r"><code>#if (!require(&quot;BiocManager&quot;, quietly = TRUE))
#    install.packages(&quot;BiocManager&quot;)

#BiocManager::install(&quot;DESeq2&quot;)


#if (!require(&quot;BiocManager&quot;, quietly = TRUE))
#    install.packages(&quot;BiocManager&quot;)

#BiocManager::install(&quot;apeglm&quot;)




library(DESeq2)
library(apeglm)

# ----------------------------
# Prepare count matrix
# ----------------------------
count_mat &lt;- as.matrix(
  union_peak_reads_count[, c(&quot;C1&quot;, &quot;C2&quot;, &quot;K1&quot;, &quot;K2&quot;)]
)

rownames(count_mat) &lt;- paste(
  union_peak_reads_count$chr,
  union_peak_reads_count$start,
  union_peak_reads_count$end,
  sep = &quot;:&quot;
)

# Ensure integer mode (important for DESeq2)
storage.mode(count_mat) &lt;- &quot;integer&quot;

# ----------------------------
# Sample metadata
# ----------------------------
coldata &lt;- data.frame(
  condition = c(&quot;C&quot;, &quot;C&quot;, &quot;K&quot;, &quot;K&quot;),
  row.names = colnames(count_mat)
)

# ----------------------------
# Create DESeq2 object
# ----------------------------
dds &lt;- DESeqDataSetFromMatrix(
  countData = count_mat,
  colData   = coldata,
  design    = ~ condition
)

# ----------------------------
# Optional but recommended: filter low counts
# ----------------------------
#keep &lt;- rowSums(counts(dds)) &gt;= 10
#dds &lt;- dds[keep, ]

# ----------------------------
# Run DESeq2
# ----------------------------
dds &lt;- DESeq(dds)

# ----------------------------
# Results: K vs C
# ----------------------------
res &lt;- results(dds, contrast = c(&quot;condition&quot;, &quot;K&quot;, &quot;C&quot;))

head(res)</code></pre>
<pre><code>log2 fold change (MLE): condition K vs C 
Wald test p-value: condition K vs C 
DataFrame with 6 rows and 6 columns
                      baseMean log2FoldChange     lfcSE      stat    pvalue
                     &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
chr1:3083877:3084023    9.3194      -1.805230  1.457376 -1.238685 0.2154620
chr1:3120283:3120658   60.5243      -0.474142  0.548569 -0.864324 0.3874098
chr1:3128676:3128999   14.6664       2.645541  1.145251  2.310009 0.0208876
chr1:3149871:3150061   19.1308      -1.721634  0.966142 -1.781968 0.0747545
chr1:3251199:3251666   44.1007      -0.808456  0.676084 -1.195794 0.2317771
chr1:3300549:3300826   21.6822      -1.639670  0.957792 -1.711926 0.0869102
                          padj
                     &lt;numeric&gt;
chr1:3083877:3084023  0.693042
chr1:3120283:3120658  0.763792
chr1:3128676:3128999  0.565293
chr1:3149871:3150061  0.629236
chr1:3251199:3251666  0.699926
chr1:3300549:3300826  0.637331</code></pre>
<pre class="r"><code># Shrink log2FC (recommended for ATAC-seq)
#res_shrunk &lt;- lfcShrink(
#  dds,
#  contrast = c(&quot;condition&quot;, &quot;K&quot;, &quot;C&quot;),
#  type = &quot;apeglm&quot;
#)

# Order by adjusted p-value
#res_shrunk &lt;- res_shrunk[order(res_shrunk$padj), ]
#
#head(res_shrunk)</code></pre>
<pre class="r"><code>library(dplyr)
library(ggplot2)


log2FC_cut &lt;- 0.38
pval_cut   &lt;- 0.05


library(dplyr)

plot_df &lt;- as.data.frame(res) %&gt;%
  mutate(
    mean_expr = baseMean,
    neglogp   = -log10(pvalue),
    Typechange = case_when(
      pvalue &lt; pval_cut &amp; log2FoldChange &gt;=  log2FC_cut  ~ &quot;Enhanced&quot;,
      pvalue &lt; pval_cut &amp; log2FoldChange &lt;= -log2FC_cut  ~ &quot;Weakened&quot;,
      TRUE                                               ~ &quot;Unchanged&quot;
    )
  )
plot_df$neglogp &lt;- pmin(plot_df$neglogp, 15)</code></pre>
<pre class="r"><code>library(ggplot2)

ggplot(plot_df, aes(x = mean_expr, y = log2FoldChange)) +
  geom_point(
    aes(color = Typechange, size = neglogp),
    alpha = 0.6
  ) +
  scale_x_log10() +
  geom_hline(yintercept = 0, color = &quot;blue&quot;, linewidth = 1) +
  geom_hline(
    yintercept = c(-log2FC_cut, log2FC_cut),
    linetype = &quot;dashed&quot;,
    color = &quot;black&quot;
  ) +
  scale_color_manual(
    values = c(
      &quot;Enhanced&quot;   = &quot;#F8766D&quot;,
      &quot;Weakened&quot;     = &quot;#00BA38&quot;,
      &quot;Unchanged&quot; = &quot;#619CFF&quot;
    )
  ) +
  scale_size_continuous(
    range = c(1, 6),
    name = &quot;-log10(pvalue)&quot;
  ) +
  labs(
    x = &quot;Log Mean expression&quot;,
    y = &quot;Log2 Fold Change&quot;,
    color = &quot;Typechange&quot;
  ) +
  theme_bw(base_size = 14) +
  theme(
    panel.grid.major = element_line(color = &quot;grey90&quot;),
    panel.grid.minor = element_blank()
  )</code></pre>
<p><img src="figure/figure1_analysis.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/autism_epigenetics/blob/5e48e7e7886700315eec5203bc459f774e73f1ba/docs/figure/figure1_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">5e48e7e</a>
</td>
<td>
han
</td>
<td>
2026-02-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/han16/autism_epigenetics/blob/85aa352d46f4a180cef7237d1179c4c86f7c4fc8/docs/figure/figure1_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">85aa352</a>
</td>
<td>
han
</td>
<td>
2026-02-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#################### normalize counts 
library(DESeq2)

count_mat &lt;- as.matrix(count_mat)

coldata &lt;- data.frame(
  row.names = colnames(count_mat),
  condition = c(&quot;C&quot;, &quot;C&quot;, &quot;K&quot;, &quot;K&quot;)
)

dds &lt;- DESeqDataSetFromMatrix(
  countData = count_mat,
  colData = coldata,
  design = ~ condition
)

# filter low-count peaks (important)
dds &lt;- dds[rowSums(counts(dds)) &gt;= 10, ]

# variance-stabilizing transform
vsd &lt;- vst(dds, blind = TRUE)

mat &lt;- assay(vsd)

mat_scaled &lt;- t(scale(t(mat)))

dds &lt;- DESeq(dds)
res &lt;- results(dds, contrast = c(&quot;condition&quot;, &quot;K&quot;, &quot;C&quot;))

# keep only peaks in matrix
res &lt;- res[rownames(mat_scaled), ]</code></pre>
<pre class="r"><code># ----------------------------
# Apply significance cutoffs
# ----------------------------
res_filt &lt;- res[
  !is.na(res$pvalue) &amp;
  res$pvalue &lt; pval_cut &amp;
  abs(res$log2FoldChange) &gt;= log2FC_cut,
]

# Order by effect size
res_filt &lt;- res_filt[
  order(abs(res_filt$log2FoldChange), decreasing = TRUE),
]

# Limit size (prevents clustering crash)
topN &lt;- min(5000, nrow(res_filt))
res_filt &lt;- res_filt[1:topN, , drop = FALSE]

# ----------------------------
# Subset matrix (CRITICAL drop=FALSE)
# ----------------------------
mat_filt &lt;- mat_scaled[rownames(res_filt), , drop = FALSE]

gained &lt;- res_filt$log2FoldChange &gt; 0
lost   &lt;- res_filt$log2FoldChange &lt; 0

mat_gained &lt;- mat_filt[gained, , drop = FALSE]
mat_lost   &lt;- mat_filt[lost,   , drop = FALSE]

mat_ordered &lt;- rbind(mat_gained, mat_lost)</code></pre>
<pre class="r"><code>library(pheatmap)

pheatmap(
  mat_ordered,
  color = colorRampPalette(c(&quot;green&quot;, &quot;white&quot;, &quot;red&quot;))(100),
  cluster_rows = TRUE,
  cluster_cols = TRUE,
  show_rownames = FALSE,
  show_colnames = TRUE,
  border_color = NA,
  gaps_row = nrow(mat_gained),
  fontsize_col = 12
)</code></pre>
<p><img src="figure/figure1_analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 26200)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] pheatmap_1.0.12             ggplot2_3.5.1              
 [3] apeglm_1.26.1               DESeq2_1.44.0              
 [5] SummarizedExperiment_1.34.0 Biobase_2.64.0             
 [7] MatrixGenerics_1.16.0       matrixStats_1.3.0          
 [9] GenomicRanges_1.56.2        GenomeInfoDb_1.40.0        
[11] IRanges_2.38.0              S4Vectors_0.42.0           
[13] BiocGenerics_0.50.0         dplyr_1.1.4                
[15] rprojroot_2.0.4            

loaded via a namespace (and not attached):
 [1] gtable_0.3.5            xfun_0.43               bslib_0.7.0            
 [4] lattice_0.22-6          numDeriv_2016.8-1.1     vctrs_0.6.5            
 [7] tools_4.4.0             generics_0.1.3          parallel_4.4.0         
[10] tibble_3.2.1            fansi_1.0.6             highr_0.10             
[13] pkgconfig_2.0.3         Matrix_1.7-0            RColorBrewer_1.1-3     
[16] lifecycle_1.0.4         GenomeInfoDbData_1.2.12 farver_2.1.2           
[19] compiler_4.4.0          stringr_1.5.1           git2r_0.33.0           
[22] munsell_0.5.1           codetools_0.2-20        httpuv_1.6.15          
[25] htmltools_0.5.8.1       sass_0.4.9              yaml_2.3.8             
[28] later_1.3.2             pillar_1.9.0            crayon_1.5.2           
[31] jquerylib_0.1.4         whisker_0.4.1           MASS_7.3-60.2          
[34] BiocParallel_1.38.0     cachem_1.0.8            DelayedArray_0.30.1    
[37] emdbook_1.3.14          abind_1.4-5             bdsmatrix_1.3-7        
[40] locfit_1.5-9.12         tidyselect_1.2.1        digest_0.6.35          
[43] mvtnorm_1.3-3           stringi_1.8.4           labeling_0.4.3         
[46] fastmap_1.1.1           grid_4.4.0              colorspace_2.1-0       
[49] cli_3.6.2               SparseArray_1.4.8       magrittr_2.0.3         
[52] S4Arrays_1.4.1          utf8_1.2.4              withr_3.0.0            
[55] UCSC.utils_1.0.0        scales_1.3.0            promises_1.3.0         
[58] rmarkdown_2.26          XVector_0.44.0          httr_1.4.7             
[61] workflowr_1.7.1         coda_0.19-4.1           evaluate_0.23          
[64] knitr_1.46              bbmle_1.0.25.1          rlang_1.1.3            
[67] Rcpp_1.0.12             glue_1.7.0              rstudioapi_0.16.0      
[70] jsonlite_1.8.8          plyr_1.8.9              R6_2.5.1               
[73] fs_1.6.4                zlibbioc_1.50.0        </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
